monado: 21.0.0 -> unstable-2022-05-28

LunNova opened the pull request:
Upstream hasn't had a version bump in 1.5 years, lots of progress since then.

~~Following [version normalization RFC](https://github.com/AndersonTorres/rfcs/blob/my-rfc/rfcs/0107-version-normalization.md).~~

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/881

gador commented on pkgs/applications/graphics/monado/default.nix:
```diff
@@ -46,25 +46,16 @@
-      sha256 = "093ymvi9ifpk4vyjcwhhci9cnscxwbv5f80xdbppcqa0j92nmkmp";
-    })
-  ];
-
```
Since removing the patches, `fetchPatch` can be removed from above

LunNova replied:
Done

gador reviewed (changes requested):
LGTM otherwise

gador commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 173907` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>monado</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
monado:
<details>

got build log for '/nix/store/kk211xnr7cvidw8h7phb3nl3c2d9hsnb-monado-21.0.0+date=2022-05-21' from 'daemon'
When evaluating attribute ‘monado’:
warning: unused-argument
Unused argument: `fetchpatch`.
Near pkgs/applications/graphics/monado/default.nix:4:3:

```
  |
4 | , fetchpatch
  |   ^
```
</details>
</details>

LunNova committed:
monado: 21.0.0 -> unstable-2022-05-28

Upstream hasn't had a version bump in 1.5 years, lots of progress since then.

monado: Apply suggestions from @SuperSandro2000's review

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

LunNova committed:
monado: set mainProgram to monado-cli

SuperSandro2000 commented on pkgs/applications/graphics/monado/default.nix:
```diff
@@ -46,25 +45,16 @@
 stdenv.mkDerivation rec {
   pname = "monado";
-  version = "21.0.0";
+  version = "21.0.0+date=2022-05-21";
```
```suggestion
  version = "unstable-2022-05-21";
```

SuperSandro2000 commented on pkgs/applications/graphics/monado/default.nix:
```diff
@@ -46,25 +45,16 @@
 
   src = fetchFromGitLab {
     domain = "gitlab.freedesktop.org";
     owner = pname;
     repo = pname;
```
```suggestion
    owner = "monado";
    repo = "monado";
```
makes it easier to get to the source url.

SuperSandro2000 commented on pkgs/applications/graphics/monado/default.nix:
```diff
@@ -130,5 +120,8 @@ stdenv.mkDerivation rec {
     license = licenses.boost;
     maintainers = with maintainers; [ expipiplus1 prusnak ];
     platforms = platforms.linux;
+    # Seemed like the most obvious choice, package also includes
+    # monado-ctl, monado-gui, monado-service
```
```suggestion
```

expipiplus1 commented:
@LunNova LGTM, happy for you (or anyone to merge), but could you please squash the commits, thanks :)

LunNova commented:
Rebased, squashed, and bumped to `unstable-2022-05-28`

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/514
